### PR TITLE
PHPORM-83: Document release process

### PR DESCRIPTION
**Pull request by alcaeus**
PHPORM-83

This PR adds a `RELEASE.md` file, which is mostly copied over from PHPLIB and adjusted to account for the minor differences in projects.

### CODE REVIEW

File: RELEASING.md [hunk 1]
```diff
@@ -0,0 +1,189 @@
+# Releasing
+
+The following steps outline the release process for both new minor versions (e.g.
+releasing the `master` branch as X.Y.0) and patch versions (e.g. releasing the
+`vX.Y` branch as X.Y.Z).
```
Comment (alcaeus) [inline RELEASING.md:5]: Current branches use a naming strategy without a leading `v` (i.e. `3.9`). In the interest of consistency, I'd change this to `v4.0` starting with the 4.0 release series.

File: RELEASING.md [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+$ git commit -m "Update composer.json X.Y.Z" composer.json
+$ git push mongodb
+```
+
+## Tag the release
+
+Create a tag for the release and push:
+
+```console
+$ git tag -a -m "Release X.Y.Z" X.Y.Z
```
Comment (alcaeus) [inline RELEASING.md:65]: Keeping `-a` for the time being, but we can consider changing this to `-s` to enforce signed tags.

File: RELEASING.md [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+Ensure that the test suite completes on supported versions of PHP.
+
+## Transition JIRA issues and version
+
+All issues associated with the release version should be in the "Closed" state
+and have a resolution of "Fixed". Issues with other resolutions (e.g.
+"Duplicate", "Works as Designed") should be removed from the release version so
+that they do not appear in the release notes.
+
+Check the corresponding ".x" fix version to see if it contains any issues that
```
[the remark this answers is not in the document]
Comment (alcaeus) [reply RELEASING.md]: Changed

File: RELEASING.md [hunk 4]
```diff
@@ -0,0 +1,189 @@
...
+4.0.1) would then be done within that branch and any development for the next
+major or minor release can continue in master.
+
+After creating a maintenance branch, the `extra.branch-alias.dev-master` field
+in the master branch's `composer.json` file should be updated. For example,
+after branching v4.0, `composer.json` in the master branch may still read:
+
+```
+"branch-alias": {
+    "dev-master": "4.0.x-dev"
```
[the remark this answers is not in the document]
Comment (alcaeus) [reply RELEASING.md:85]: Added

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -52,4 +52,9 @@ If the project maintainer has any additional requirements, you will find them li
 
 - **Send coherent history** - Make sure each individual commit in your pull request is meaningful. If you had to make multiple intermediate commits while developing, please [squash them](https://www.git-scm.com/book/en/v2/Git-Tools-Rewriting-History#Changing-Multiple-Commit-Messages) before submitting.
 
-**Happy coding**!
\ No newline at end of file
+**Happy coding**!
```
Comment (jmikola) [inline CONTRIBUTING.md:55]:
```suggestion
Happy coding!

```

Bold formatting above makes sense to distinguish the text from the remainder but I don't think it adds anything here. Let me know if this is enough bike-shedding for you. I can go deeper.

File: RELEASING.md (hunk 1 above)
Comment (jmikola) [reply RELEASING.md:5]: IIRC, "v" prefixing was used to clearly distinguish branches from tags. I assume the ".x" suffix used by Laravel and Doctrine accomplishes the same thing.

That said, our tags always consist of three parts (e.g. `X.Y.Z`), so the real benefit here might just be removing the need to context switch between our various repositories.

File: RELEASING.md [hunk 6]
```diff
@@ -0,0 +1,190 @@
...
+
+## Transition JIRA issues and version
+
+All issues associated with the release version should be in the "Closed" state
+and have a resolution of "Fixed". Issues with other resolutions (e.g.
+"Duplicate", "Works as Designed") should be removed from the release version so
+that they do not appear in the release notes.
+
+Check the corresponding "laravel-*.x" fix version to see if it contains any
+issues that  are resolved as "Fixed" and should be included in this release
```
Comment (jmikola) [inline RELEASING.md:28]:
```suggestion
issues that are resolved as "Fixed" and should be included in this release
```

File: RELEASING.md [hunk 7]
```diff
@@ -0,0 +1,190 @@
...
+
+```markdown
+**Thanks**
+
+Thanks for our community contributors for this release:
+
+ * [$CONTRIBUTOR_NAME](https://github.com/$GITHUB_USERNAME)
+```
+
+Release announcements should also be posted in the [MongoDB Product & Driver Announcements: Driver Releases](https://mongodb.com/community/forums/tags/c/announcements/driver-releases/110/php) forum and shared on Twitter.
```
Comment (jmikola) [inline RELEASING.md:190]: Do you want to remove this? Or is it consistent with how other languages announce their extra-driver releases (e.g. Ruby announcing Mongoid)?

**Review by jmikola — APPROVED** [review]
A few suggestions and no objections.

File: RELEASING.md (hunk 2 above)
Comment (GromNaN) [reply RELEASING.md:66]: I'm OK with signing the tag.

File: RELEASING.md (hunk 1 above)
Comment (GromNaN) [reply RELEASING.md:5]: Okay, I guess it doesn't really matter. Consistency across our projects is the most important thing.

Comment (alcaeus) [reply RELEASING.md:5]: Correct, that was the main motivation to do this.

File: RELEASING.md (hunk 7 above)
Comment (alcaeus) [reply RELEASING.md:190]: Ruby does indeed post these for Mongoid as well. IIRC the announcements should cover all products maintained by the DBX team, hence the inclusion here.

File: RELEASING.md (hunk 6 above)
Comment (alcaeus) [reply RELEASING.md:28]: Fixed

File: CONTRIBUTING.md (hunk 5 above)
Comment (alcaeus) [reply CONTRIBUTING.md:55]: Done - I didn't want to bother to update the contribution docs too much here, but I believe a second pass in the near future is in order.

File: RELEASING.md (hunk 2 above)
Comment (jmikola) [reply RELEASING.md:66]: We already have a DRIVERS epic and tracking tickets for this. See [PHPC-1584](https://jira.mongodb.org/browse/PHPC-1584). I'd propose creating a PHPORM ticket and linking that up.

**Pull request merged**